### PR TITLE
STORM-721. Storm UI server should support SSL.

### PR DESCRIPTION
**Pull request by harshach**

### CODE REVIEW

File: storm-core/src/jvm/backtype/storm/Config.java [hunk 1]
```diff
@@ -567,6 +567,39 @@
     public static final Object UI_HEADER_BUFFER_BYTES_SCHEMA = Number.class;
 
     /**
+     * This port is used by Storm DRPC for receiving HTTPS (SSL) DPRC requests from clients.
+     */
+    public static final String UI_HTTPS_PORT = "ui.https.port";
```
Comment (caofangkun) [inline storm-core/src/jvm/backtype/storm/Config.java]: Will be better if you could  modify file  
```
storm-core/conf/defaults.yaml  
```
and give these newly added configs default values.

**Comment by harshach** [comment]
@caofangkun this is not a manadatory fields as it its optional to configure ssl so no need for defaults here.

**Comment by harshach** [comment]
@ptgoetz @Parth-Brahmbhatt @revans2 appreciate any feedback on this PR>

File: storm-core/src/clj/backtype/storm/daemon/drpc.clj [hunk 2]
```diff
@@ -225,7 +225,13 @@
               https-port (int (conf DRPC-HTTPS-PORT))
               https-ks-path (conf DRPC-HTTPS-KEYSTORE-PATH)
               https-ks-password (conf DRPC-HTTPS-KEYSTORE-PASSWORD)
-              https-ks-type (conf DRPC-HTTPS-KEYSTORE-TYPE)]
+              https-ks-type (conf DRPC-HTTPS-KEYSTORE-TYPE)
+              https-key-password (conf DRPC-HTTPS-KEY-PASSWORD)
+              https-ts-path (conf DRPC-HTTPS-TRUSTSTORE-PATH)
+              https-ts-password (conf DRPC-HTTPS-TRUSTSTORE-PASSWORD)
+              https-ts-type (conf DRPC-HTTPS-TRUSTSTORE-TYPE)
+              https-want-client-auth (conf DRPC-HTTPS-WANT-CLIENT-AUTH)
```
Comment (Parth-Brahmbhatt) [inline storm-core/src/clj/backtype/storm/daemon/drpc.clj]: the documentation for DRPC-HTTPS-WANT-CLIENT-AUTH and DRPC-HTTPS-NEED-CLIENT-AUTH is missing from SECURITY.md.

File: storm-core/src/clj/backtype/storm/ui/core.clj [hunk 3]
```diff
@@ -1078,10 +1078,32 @@
...
+                          :filter-params (conf UI-FILTER-PARAMS)}]
+          https-port (if (not-nil? (conf UI-HTTPS-PORT)) (conf UI-HTTPS-PORT) 0)
+          https-ks-path (conf UI-HTTPS-KEYSTORE-PATH)
+          https-ks-password (conf UI-HTTPS-KEYSTORE-PASSWORD)
+          https-ks-type (conf UI-HTTPS-KEYSTORE-TYPE)
+          https-key-password (conf UI-HTTPS-KEY-PASSWORD)
+          https-ts-path (conf UI-HTTPS-TRUSTSTORE-PATH)
+          https-ts-password (conf UI-HTTPS-TRUSTSTORE-PASSWORD)
+          https-ts-type (conf UI-HTTPS-TRUSTSTORE-TYPE)
+          https-want-client-auth (conf UI-HTTPS-WANT-CLIENT-AUTH)
```
Comment (Parth-Brahmbhatt) [inline storm-core/src/clj/backtype/storm/ui/core.clj]: Same here, we should add the documentation to SECURITY.MD.

**Comment by Parth-Brahmbhatt** [comment]
I am +1 once SECURITY.MD has documentation for all the configs.

**Comment by harshach** [comment]
Thanks for the review @Parth-Brahmbhatt  . Updated the security.md .

**Comment by Parth-Brahmbhatt** [comment]
+1.

**Pull request merged**